### PR TITLE
feat: blocking setProvider

### PR DESCRIPTION
**Pull request by thomaspoignant**
## This PR

Following this [OFEP](https://github.com/open-feature/ofep/blob/main/011-OFEP-sdk-wait-provider-ready.md) adding a **SHOULD** to the spec to have a blocking `setProvider`.

### CODE REVIEW

File: specification/sections/01-flag-evaluation.md [hunk 1]
```diff
@@ -105,6 +105,22 @@ The name is a logical identifier for the client.
 
 Clients may be created in critical code paths, and even per-request in server-side HTTP contexts. Therefore, in keeping with the principle that OpenFeature should never cause abnormal execution of the first party application, this function should never throw. Abnormal execution in initialization should instead occur during provider registration.
 
+#### Requirement 1.1.8
+
+> The `API` **SHOULD** provide functions to set a provider and wait for the `initialize` function to return or throw.
+
+The `API` will ensure that the provider is ready or in error before being able to initializing the client.
```
Comment (toddbaert) [inline specification/sections/01-flag-evaluation.md:112]:
```suggestion
This function not only sets the provider, but ensures that the provider is ready (or in error) before returning or settling.
```

File: specification/sections/01-flag-evaluation.md [hunk 2]
```diff
@@ -105,6 +105,22 @@ The name is a logical identifier for the client.
...
+
+```typescript
+// default client
+await Openfeature.setProviderAndWait(myprovider);
+const client = Openfeature.getClient();
+
+// named client
+await Openfeature.setProviderAndWait('client-name', myprovider);
+const client = Openfeature.getClient('client-name');
+```
```
Comment (toddbaert) [inline specification/sections/01-flag-evaluation.md:122]: In general, for consistency, I've tried to stick to Java in the spec. I created [this](https://github.com/open-feature/spec/issues/187) a while ago to try to make that more consistent. It's up to you, but I'd recommend making this look "Java-ish" and add a comment saying `// this method blocks` or something like that.

File: specification/sections/01-flag-evaluation.md [hunk 3]
```diff
@@ -105,6 +105,22 @@ The name is a logical identifier for the client.
...
+
```
Comment (toddbaert) [inline specification/sections/01-flag-evaluation.md:123]:
```suggestion

Though it's possible to use [events](./05-events.md) to await provider readiness, such functions can make things simpler for `application authors` and `integrators`.

```

File: specification/sections/01-flag-evaluation.md (hunk 2 above)
Comment (thomaspoignant) [reply specification/sections/01-flag-evaluation.md]: I've seen a bunch of typescript examples that's why I went for it. I've changed it to Java.

**Review by toddbaert — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

**Review by justinabrahms — APPROVED** [review]

**Pull request merged**